plugins/hello_world_multi_step.py

```diff
@@ -0,0 +1,373 @@
+        HelloWorldParametersSchema(
+            partial=True, unknown=EXCLUDE, validate_errors_as_result=True
+        ),
+        location="query",
+        required=False,
+    )
+    @HELLO_MULTI_BLP.require_jwt("jwt", optional=True)
+    def get(self, errors):
+        """Return the micro frontend."""
+        return self.render(request.args, errors)
+
+    @HELLO_MULTI_BLP.html_response(
+        HTTPStatus.OK, description="Micro frontend of the hello world plugin."
+    )
+    @HELLO_MULTI_BLP.arguments(
+        HelloWorldParametersSchema(
+            partial=True, unknown=EXCLUDE, validate_errors_as_result=True
+        ),
+        location="form",
+        required=False,
+    )
+    @HELLO_MULTI_BLP.require_jwt("jwt", optional=True)
+    def post(self, errors):
+        """Return the micro frontend with prerendered inputs."""
+        return self.render(request.form, errors)
+
+    def render(self, data: Mapping, errors: dict):
+        schema = HelloWorldParametersSchema()
+        return Response(
+            render_template(
+                "hello_template.html",
+                name=HelloWorld.instance.name,
+                version=HelloWorld.instance.version,
+                schema=schema,
+                values=data,
+                errors=errors,
+                process=url_for(f"{HELLO_MULTI_BLP.name}.ProcessView"),
+                example_values=url_for(
+                    f"{HELLO_MULTI_BLP.name}.MicroFrontend", **self.example_inputs
+                ),
+            )
+        )
+
+
+@HELLO_MULTI_BLP.route("/process/")
+class ProcessView(MethodView):
+    """Start a long running processing task."""
+
+    @HELLO_MULTI_BLP.arguments(
+        HelloWorldParametersSchema(unknown=EXCLUDE), location="form"
+    )
+    @HELLO_MULTI_BLP.response(HTTPStatus.OK, TaskResponseSchema())
+    @HELLO_MULTI_BLP.require_jwt("jwt", optional=True)
+    def post(self, arguments):
+        """Start the demo task."""
+        db_task = ProcessingTask(
+            task_name="hello-world-multi-step-plugin", parameters=dumps(arguments)
+        )  # TODO: change ProcessingTask class name and task_name???
+        db_task.save(commit=True)
+
+        # set previous step to cleared
+        # nothing to do (no previous step)
+
+        # next step
+        step_id = "step1"
+        href = url_for(f"{HELLO_MULTI_BLP.name}.Step1Frontend", db_id=db_task.id)
+        ui_href = url_for(f"{HELLO_MULTI_BLP.name}.Step1View", db_id=db_task.id)
+
+        # all tasks need to know about db id to load the db entry
+        task: chain = preprocessing_task.s(db_id=db_task.id) | add_step.s(
+            db_id=db_task.id, step_id=step_id, href=href, ui_href=ui_href, prog_value=50
+        )
+
+        # save errors to db
+        task.link_error(save_task_error.s(db_id=db_task.id))
+        result: AsyncResult = task.apply_async()
+
+        # note that task_id has to be updated in each step
+        db_task.task_id = result.id
```

Comment:
That means that we don't use the result of task.apply_async anymore, correct? 
Instead of 
`result: AsyncResult = task.apply_async()

# note that task_id has to be updated in each step
db_task.task_id = result.id`

we would simply write
`task.apply_async()`

Reply:
That should be the case, yes